test/AccessControl.js

```diff
@@ -3,24 +3,32 @@ import EvmError from "./helpers/EVMThrow";
 import { eventValue } from "./helpers/events";
 import { TriState } from "./helpers/triState";
 import roles from "./helpers/roles";
+import { saveBlockchain, restoreBlockchain } from "./helpers/evmCommands";
 
 const RoleBasedAccessControl = artifacts.require("RoleBasedAccessControl");
 const TestAccessControlTruffleMixin = artifacts.require(
   "TestAccessControlTruffleMixin"
 );
 
 contract("AccessControl", ([accessController, owner1, owner2]) => {
+  let snapshot;
   let accessControl;
   let accessControlled;
   let exampleRole;
 
-  beforeEach(async () => {
+  before(async () => {
     accessControl = await RoleBasedAccessControl.new();
     accessControlled = await TestAccessControlTruffleMixin.new(
       accessControl.address
     );
 
     exampleRole = roles.example;
+    snapshot = await saveBlockchain();
+  });
+
+  beforeEach(async () => {
+    await restoreBlockchain(snapshot);
+    snapshot = await saveBlockchain();
```

Comment:
Is this because you cannot restore twice the same snapshot? weeeiiirrddd

Reply:
Yeah, seems so. I think it's more of a rollback operation than a restore.